### PR TITLE
LPD-30142 Primary key from modifiable system objects have to starts with l_

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-30142

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java [hunk 1]
```diff
@@ -1232,6 +1232,8 @@ public void testAddSystemObjectDefinition() throws Exception {
 				ObjectDefinitionConstants.
 					EXTERNAL_REFERENCE_CODE_PREFIX_SYSTEM_OBJECT_DEFINITION));
 		Assert.assertEquals("/test", objectDefinition.getRESTContextPath());
+		Assert.assertEquals(
+			"l_testId", objectDefinition.getPKObjectFieldName());
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java:1236]: @p-albuquerque why did you put the `getPKObjectFieldName` after the `getRESTContextPath`?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 2]
```diff
@@ -399,6 +402,12 @@ public void testAddSystemObjectRelationship() throws Exception {
...
 
+		_testAddObjectRelationshipManyToMany(
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			_modifiableSystemObjectDefinition, _objectDefinition1, false);
+		_testAddObjectRelationshipManyToMany(
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			_modifiableSystemObjectDefinition, _objectDefinition1, true);
 		_testAddObjectRelationshipManyToMany(
 			ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
 			_objectDefinition1, _systemObjectDefinition2, false);
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:413]: @p-albuquerque the `DELETION_TYPE_PREVENT` should be placed after the `DELETION_TYPE_DISASSOCIATE` to follow the existing pattern

File: modules/apps/object/object-service/bnd.bnd [hunk 3]
```diff
@@ -2,6 +2,6 @@ Bundle-Name: Liferay Object Service
 Bundle-SymbolicName: com.liferay.object.service
 Bundle-Version: 1.0.201
 Liferay-Client-Extension-Batch: com/liferay/object/internal/batch
-Liferay-Require-SchemaVersion: 9.2.2
+Liferay-Require-SchemaVersion: 9.2.3
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/bnd.bnd:5]: @p-albuquerque if you are changing a column name, then it's a major change. Please see this [article](https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Major-changes).

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_3/ObjectDefinitionUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					StringBundler.concat(
+						"update ObjectDefinition set ",
+						"pkObjectFieldDBColumnName = ?, pkObjectFieldName = ? ",
+						"where objectDefinitionId = ?"));
+			ResultSet resultSet = preparedStatement1.executeQuery()) {
+
+			while (resultSet.next()) {
+				String pkObjectFieldDBColumnName = resultSet.getString(
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_3/ObjectDefinitionUpgradeProcess.java:41]: @p-albuquerque I would rename it to `oldPKObjectFieldDBColumnName`

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_3/ObjectDefinitionUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+				preparedStatement2.setString(1, newPKObjectFieldDBColumnName);
+
+				preparedStatement2.setString(
+					2,
+					StringUtil.replaceFirst(
+						resultSet.getString("pkObjectFieldName"), "c_", "l_"));
+
+				preparedStatement2.setLong(
+					3, resultSet.getLong("objectDefinitionId"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_3/ObjectDefinitionUpgradeProcess.java:55]: @p-albuquerque please remove the empty line

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_3/ObjectDefinitionUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+				String pkObjectFieldDBColumnDefinition =
+					newPKObjectFieldDBColumnName + " LONG not null";
+
+				alterColumnName(
+					resultSet.getString("dbTableName"),
+					pkObjectFieldDBColumnName, pkObjectFieldDBColumnDefinition);
+
+				alterColumnName(
+					resultSet.getString("dbTableName") + "_x",
+					pkObjectFieldDBColumnName, pkObjectFieldDBColumnDefinition);
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_3/ObjectDefinitionUpgradeProcess.java:68]: @p-albuquerque I would inline and also remove the empty line:

```
alterColumnName(
	resultSet.getString("dbTableName"),
	oldPKObjectFieldDBColumnName, newPKObjectFieldDBColumnName + " LONG not null");
alterColumnName(
	resultSet.getString("dbTableName") + "_x",
	oldPKObjectFieldDBColumnName, newPKObjectFieldDBColumnName + " LONG not null");
```

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_3/test/ObjectDefinitionUpgradeProcessTest.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+				objectDefinition.getObjectDefinitionId());
+
+		String pkObjectFieldDBColumnName =
+			objectDefinition.getPKObjectFieldDBColumnName();
+
+		Assert.assertTrue(pkObjectFieldDBColumnName.startsWith("c_"));
+
+		String pkObjectFieldName = objectDefinition.getPKObjectFieldName();
+
+		Assert.assertTrue(pkObjectFieldName.startsWith("c_"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_3/test/ObjectDefinitionUpgradeProcessTest.java:84]: @p-albuquerque simplify:

```
Assert.assertTrue(
	StringUtil.startsWith(
		objectDefinition.getPKObjectFieldDBColumnName(), "c_"));
Assert.assertTrue(
	StringUtil.startsWith(
		objectDefinition.getPKObjectFieldName(), "c_"));
```

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_3/test/ObjectDefinitionUpgradeProcessTest.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+			objectDefinition.getObjectDefinitionId());
+
+		pkObjectFieldDBColumnName =
+			objectDefinition.getPKObjectFieldDBColumnName();
+
+		Assert.assertTrue(pkObjectFieldDBColumnName.startsWith("l_"));
+
+		pkObjectFieldName = objectDefinition.getPKObjectFieldName();
+
+		Assert.assertTrue(pkObjectFieldName.startsWith("l_"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_3/test/ObjectDefinitionUpgradeProcessTest.java:107]: @p-albuquerque simplify:

```
Assert.assertTrue(
	StringUtil.startsWith(
		objectDefinition.getPKObjectFieldDBColumnName(), "l_"));
Assert.assertTrue(
	StringUtil.startsWith(
		objectDefinition.getPKObjectFieldName(), "l_"));
```

Comment (carolmariaabb) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_2_3/test/ObjectDefinitionUpgradeProcessTest.java:107]: @p-albuquerque you can also create a private method to reuse code:

```
private void _assertObjectDefinition(ObjectDefinition objectDefinition, String prefix)
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java:1236]: Good question

**Comment by p-albuquerque** [comment]
@carolmariaabb two questions:
I think a new name for the private method in the upgrade test, what do you think about it? Is it better more verbose name, or just a more generic one?
Is it okay keep "old" and "new" prefix for the variables in the upgrade process? Or is it better remove the "new" prefix?

**Comment by carolmariaabb** [comment]
> @carolmariaabb two questions: I thought a new name for the private method to perform the assert in the upgrade test, what do you think about it? Is it better more verbose name, or just a more generic one? Is it okay keep "old" and "new" prefix for the variables in the upgrade process? Or is it better remove the "new" prefix?

We can keep your suggestion and see what Brians says.
Regarding the second question, if we have the `old` prefix, I think it makes sense to keep the `new` prefix.

**Review by carolmariaabb — APPROVED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:object

**Comment by p-albuquerque** [comment]
ci:report:19226571

**Comment by p-albuquerque** [comment]
ci:forward

**Comment by p-albuquerque** [comment]
Directly sent: https://github.com/brianchandotcom/liferay-portal/pull/154478

**Pull request closed without merging**